### PR TITLE
Support `Function` references in `Alias` resource

### PR DESCRIPTION
**Pull request by Vandita2020**
Issue #, if available: 

Description of changes:
Adding functionality to the lambda controller to be able to reference a `Function` from an `Alias` resource using resource references.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/tests/test_alias.py [hunk 1]
```diff
@@ -80,18 +80,19 @@ def lambda_function():
...
         replacements = REPLACEMENT_VALUES.copy()
         replacements["AWS_REGION"] = get_region()
         replacements["ALIAS_NAME"] = resource_name
-        replacements["FUNCTION_NAME"] = lambda_function_name
+        replacements["FUNCTION_REF_NAME"] = function_resource_name
         replacements["FUNCTION_VERSION"] = "$LATEST"
 
         # Load alias CR
-        resource_data = load_lambda_resource(
+        resource_data = load_lambda_resource(
```
Comment (jaypipes) [inline test/e2e/tests/test_alias.py:95]: looks like an extraneous whitespace found its way in here :)

**Review by jaypipes — CHANGES_REQUESTED** [review]
Hi @Vandita2020! Thanks very much for this contribution! May I ask that you update the commit message and pull request summary to mention that you are adding functionality to the lambda controller to be able to reference a `Function` from an `Alias` resource using resource references?

Thank you!
-jay

File: test/e2e/tests/test_alias.py [hunk 2]
```diff
@@ -136,3 +136,63 @@ def test_smoke(self, lambda_client, lambda_function):
...
+
+        replacements = REPLACEMENT_VALUES.copy()
+        replacements["AWS_REGION"] = get_region()
+        replacements["ALIAS_NAME"] = resource_name
+        replacements["FUNCTION_REF_NAME"] = function_resource_name
+        replacements["FUNCTION_VERSION"] = "$LATEST"
+
+        # Load alias CR
+        resource_data = load_lambda_resource( 
+            "alias",
```
Comment (A-Hilaly) [inline test/e2e/tests/test_alias.py:154]: Did you mean `alias-ref`? :)

File: generator.yaml [hunk 3]
```diff
@@ -93,4 +96,4 @@ resources:
       ignore: true
     fields:
       FunctionName:
-        is_primary_key: true
\ No newline at end of file
+        is_primary_key: true
```
Comment (A-Hilaly) [inline generator.yaml:99]:
```suggestion
        is_primary_key: true
```

File: pkg/resource/alias/references.go [hunk 4]
```diff
@@ -36,17 +43,95 @@ func (rm *resourceManager) ResolveReferences(
...
+	}
+	return &resource{ko}, err
 }
 
 // validateReferenceFields validates the reference field and corresponding
 // identifier field.
 func validateReferenceFields(ko *svcapitypes.Alias) error {
+	if ko.Spec.FunctionRef != nil && ko.Spec.FunctionName != nil {
+		return ackerr.ResourceReferenceAndIDNotSupportedFor("FunctionName", "FunctionRef")
+	}
```
Comment (A-Hilaly) [inline pkg/resource/alias/references.go:69]: Remember the case of users providing `FunctionName` and `FunctionRef`? Looks like it's handled here

**Review by A-Hilaly — COMMENTED** [review]
Nice start on this @Vandita2020 ! I left two tiny comments inline.

File: test/e2e/tests/test_alias.py (hunk 1 above)
Comment (Vandita2020) [reply test/e2e/tests/test_alias.py:94]: Oops! corrected it. Thanks.

File: test/e2e/tests/test_alias.py (hunk 2 above)
Comment (Vandita2020) [reply test/e2e/tests/test_alias.py:154]: Yeah, changed it. Thanks!

File: pkg/resource/alias/references.go (hunk 4 above)
Comment (Vandita2020) [reply pkg/resource/alias/references.go:69]: Yeah, I see.

**Comment by A-Hilaly** [comment]
Nice work on this, thank you @Vandita2020 !
/lgtm

File: apis/v1alpha1/generator.yaml [hunk 5]
```diff
@@ -93,4 +96,4 @@ resources:
       ignore: true
     fields:
       FunctionName:
-        is_primary_key: true
\ No newline at end of file
+        is_primary_key: true
```
Comment (A-Hilaly) [inline apis/v1alpha1/generator.yaml:99]:
```suggestion
        is_primary_key: true
```

**Pull request merged**